react/ProductDescription.tsx

```diff
@@ -55,14 +66,39 @@ function ProductDescription(props: PropsWithChildren<Props>) {
   const { handles } = useCssHandles(CSS_HANDLES, { classes: props.classes })
   const intl = useIntl()
   const { product } = useProduct()
+  const { isMobile } = useDevice()
 
+  const { collapseContent = true, title, customCollapse = false } = props
   const description = props.description ?? product?.description
+  const [isCustomCollapsed, setCustomCollapsed] = useState(customCollapse)
+
+  const limitCharacters = useMemo(() => {
+    let {
+      desktopLimitCharacters = defaultDesktopCharacters,
+      mobileLimitCharacters = defaultMobileCharacters,
+    } = props
```

Comment:
Move these variables out of the hook so they only execute when they change. Currently, if any prop change, it's going to run the memo again.